shenyu-register-center/shenyu-register-client/shenyu-register-client-nacos/src/main/java/org/apache/shenyu/register/client/nacos/NacosClientRegisterRepository.java

```diff
@@ -157,10 +157,13 @@ private synchronized void registerConfig(final String rpcType,
         String configName = RegisterPathConstants.buildServiceConfigPath(rpcType, contextPath);
         try {
             final String defaultGroup = NacosPathConstants.GROUP;
-            configService.publishConfig(configName, defaultGroup, GsonUtils.getInstance().toJson(metadataCache));
+            if (configService.publishConfig(configName, defaultGroup, GsonUtils.getInstance().toJson(metadataCache))) {
+                LOGGER.info("register metadata success: {}", metadata.getRuleName());
```

Comment:
When will it be false?

Reply:
nacos forbids registration
so when registration fails 
 like iuuse#2710  
registration fails , but out sucess

Reply:
Maybe nacos enable [authentication](https://nacos.io/zh-cn/docs/auth.html), add username and password in the configuration, then try it again.

Reply:
However, if the registration fails, the registration success log is still output.I think this is also a problem

Reply:
Throw exception when register fail.

Reply:
will not throw exception，is only false.
look iuuse#2710  log。
output  `register metadata success `  。But actually the registration failed

Reply:
I mean you should throw exception here.